### PR TITLE
Created checks for row and col attributes specified in the $extra par…

### LINKED ISSUE #3412: Bug: form_textarea helper row and col defaults not overwriting when defined as $extra
**Describe the bug**
The default text area `row="10"` and `col="40"` is not overwritten with the value provided, either as an array or string, in the `$extra` parameter. 

**CodeIgniter 4 version**
CI 4.0.4 (master)

**Affected module(s)**
Form Helper

**Expected behavior, and steps to reproduce if appropriate**
Currently `form_textarea('demo', '', ['rows'=>'4', 'cols'=>'10'])` and `form_textarea('demo', '', 'rows="4" cols="10"')` would both output `<textarea name="mailingaddress" cols="40" rows="10"  rows="4" cols="10"></textarea>`.

**Context**
 - OS: macOS Catalina 10.15.5
 - Web server Apache 2.2.34
 - PHP version 7.4.2

### PR DESCRIPTION
**Pull request by durbintl**
Fixes #3412 

Created checks for row and col attributes specified in the $extra parameter for the form_textarea form helper.

**Checklist:**
- [X] Securely signed commits
- [ ] Component(s) with PHPdocs
- [x] Unit testing, with >80% coverage
- [ ] User guide updated
- [X] Conforms to style guide

### CODE REVIEW

**Comment by michalsn** [comment]
This looks good, but could you add some tests for this?

It's not complicated. You can look at how tests are made here: https://github.com/codeigniter4/CodeIgniter4/blob/develop/tests/system/Helpers/FormHelperTest.php#L314 and add one more to cover your changes.

**Comment by durbintl** [comment]
@michalsn Yes! I'll take care of that. I forgot, sorry. Thank's for the reminder.

**Comment by durbintl** [comment]
@michalsn All done!

File: system/Helpers/form_helper.php [hunk 1]
```diff
@@ -310,7 +310,16 @@ function form_textarea($data = '', string $value = '', $extra = ''): string
 			unset($data['value']); // textareas don't use the value attribute
 		}
 
-		return '<textarea ' . parse_form_attributes($data, $defaults) . stringify_attributes($extra) . '>'
+		// Unsets default rows and cols if defined in extra field as array or string.
+		if ((is_array($extra) && array_key_exists('rows', $extra)) || (is_string($extra) && strpos(strtolower(preg_replace('/\s+/', '', $extra)), 'rows=') !== FALSE)) {
```
Comment (michalsn) [inline system/Helpers/form_helper.php:314]:
```suggestion
		if ((is_array($extra) && array_key_exists('rows', $extra)) || (is_string($extra) && strpos(strtolower(preg_replace('/\s+/', '', $extra)), 'rows=') !== FALSE))
		{
```

File: tests/system/Helpers/FormHelperTest.php [hunk 2]
```diff
@@ -311,10 +311,11 @@ public function testFormUpload()
 	}
 
 	// ------------------------------------------------------------------------
-	public function testFormTextarea()
+	public function testForm
+		()
```
Comment (michalsn) [inline tests/system/Helpers/FormHelperTest.php:315]:
```suggestion
	public function testFormTextarea()
```

File: system/Helpers/form_helper.php [hunk 3]
```diff
@@ -310,7 +310,16 @@ function form_textarea($data = '', string $value = '', $extra = ''): string
...
+			unset($defaults['rows']);
+		}
+
+		if ((is_array($extra) && array_key_exists('cols', $extra)) || (is_string($extra) && strpos(strtolower(preg_replace('/\s+/', '', $extra)), 'cols=') !== FALSE)) {
```
Comment (michalsn) [inline system/Helpers/form_helper.php:318]:
```suggestion
		if ((is_array($extra) && array_key_exists('cols', $extra)) || (is_string($extra) && strpos(strtolower(preg_replace('/\s+/', '', $extra)), 'cols=') !== FALSE))
		{
```

**Pull request merged**